### PR TITLE
Bio1 2019 q2

### PR DESCRIPTION
**Pull request by AG-88301**

### CODE REVIEW

File: solutions/bio1/Trail/index.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      </SText>
+
+      <STitle>Solution</STitle>
+      <SText>
+        To keep track of the trail, we can use a hashmap, storing the last time a cell was visited.
+        Unvisited cells can be stored as $-{brace("\\text{inf}")}$, where ${brace("\\text{inf}")}$ is a large number.
+        We can simulate each move by keeping track of the direction and position of the explorer.
+        To check if a cell contains a trail, we simply need to compare the current time, with the time the trail was last visited.
+        Formally, if the current time is $j$, and the last time the cell was visited was $i$, 
+        the trail still exists if $j - i {brace("~<~")} t$.
```
Comment (myst-6) [inline solutions/bio1/Trail/index.tsx:20]: the brace thing with `~<~` is a bit strange. You can just use \lt for this.

Comment (myst-6) [reply solutions/bio1/Trail/index.tsx:20]: I think it's good to clean up the code by changing this. Don't forget to remove the unused brace import.

**Review by myst-6 — CHANGES_REQUESTED** [review]
I think it's good. Just nitpicking slightly

File: solutions/bio1/Trail/index.tsx (hunk 1 above)
Comment (AG-88301) [reply solutions/bio1/Trail/index.tsx:20]: brace is still used by inf

**Comment by macaquedev** [comment]
Also you shouldn't say "inf, where inf is a large number", and then proceed to use float("inf") in your code...!! Do you know that float('inf') is not in fact "just a large number"?

**Comment by AG-88301** [comment]
> Also you shouldn't say "inf, where inf is a large number", and then proceed to use float("inf") in your code...!! Do you know that float('inf') is not in fact "just a large number"?

how else do i describe it... an overridden float that will always be bigger than every other number

**Comment by macaquedev** [comment]
say "set it to inf, or some very large number

**Comment by AG-88301** [comment]
> say "set it to inf, or some very large number

so ur saying make it
```
Unvisited cells can be stored as $-{brace("\\text{inf}")}$.
```
...?

**Comment by myst-6** [comment]
> > say "set it to inf, or some very large number
> 
> so ur saying make it
> 
> ```
> Unvisited cells can be stored as $-{brace("\\text{inf}")}$.
> ```
> 
> ...?

Is there any reason not to just use $-1$?

**Comment by macaquedev** [comment]
@AG-88301 make it:
"Unvisited cells can be stored as -inf, or some other very large negative number"

is there any way to avoid -inf? -1 comes to mind....

**Comment by AG-88301** [comment]
-1 won't work... I could use -t but effort and requires more explaining

**Review by myst-6 — APPROVED** [review]
I think it's good.

**Review by macaquedev — CHANGES_REQUESTED** [review]
sorry you're wrong @myst-6 . i'm not the only one who doesn't notice stuff!

image
missing full stop LOL

**Comment by myst-6** [comment]
> sorry you're wrong @myst-6 . i'm not the only one who doesn't notice stuff!
> 
> image missing full stop LOL

oops!!

**Review by macaquedev — APPROVED** [review]
good

**Pull request merged**